ui/lib/pki/addon/routes/overview.js

```diff
@@ -1,3 +1,42 @@
 import Route from '@ember/routing/route';
+import { inject as service } from '@ember/service';
+import RSVP from 'rsvp';
 
-export default class PkiOverviewRoute extends Route {}
+export default class PkiOverviewRoute extends Route {
+  @service secretMountPath;
+  @service auth;
+  @service store;
+
+  get win() {
+    return this.window || window;
+  }
+
+  hasConfig() {
+    const endpoint = `${this.win.origin}/v1/${this.secretMountPath.currentPath}/issuers?list=true`;
+    return this.auth
+      .ajax(endpoint, 'GET', {})
+      .then(() => true)
+      .catch(() => false);
+  }
+
+  async model() {
+    return RSVP.hash({
+      hasConfig: this.hasConfig(),
+      engine: this.store
+        .query('secret-engine', {
+          path: this.secretMountPath.currentPath,
+        })
+        .then((model) => {
+          if (model) {
+            return model.get('firstObject');
+          }
+        }),
```

Comment:
I'm not sure if you need this, but since this is in a `hash` you may need a `catch` statement if any promises fail

Reply:
If it fails the page will load the error state, which is what we want